core/controllers/topic_editor.py

```diff
@@ -78,7 +78,6 @@ def post(self, topic_id):
         story = story_domain.Story.create_default_story(
             new_story_id, title=title)
         story_services.save_new_story(self.user_id, story)
-        topic_services.add_canonical_story(self.user_id, topic_id, new_story_id)
```

Comment:
I'm not sure I understand what's happening here, how come we don't add the story to the topic?

Reply:
For this and the comment above, doing this in the backend meant that the local topic object was not updated, so if the user deletes a story,  for example, the frontend list would be updated only after a page refresh. So, to fix this discrepancy, I removed all such updates from the backend, and made it through frontend calls. 
Though, this can also be done by keeping the backend as it is and forcing a page refresh after each delete, would that be fine?

Reply:
I think this is a problem, right? Suppose I go to the topic editor, create a story, and then discard changes. What happens to the story? Ditto for deletion.

So I think you want to force the changelist queue to be empty _before_ create/delete, and then when the create/delete happens, treat it as an atomic operation that gets saved immediately.

Reply:
Yeah, add story is treated as an atomic change in that way. When create story is clicked, the story is added to the topic and the change is saved soon after, after which the user is directed to story editor. So, the user simply cannot discard an add canonical story change. When the user comes back to the topic editor, there won't be any pending changes, so they can then delete the story if they want to.